Frontend/src/pages/register/components/Processing.jsx

```diff
@@ -0,0 +1,48 @@
+import { useQuery } from '@tanstack/react-query'
+import React, { useState } from 'react'
+import { Message } from 'semantic-ui-react'
+import { pollRegistrations } from '../../../api/registration/get/poll_registrations'
+
+const REFETCH_INTERVAL = 3000
+
+export default function Processing({ processing, setProcessing }) {
+  const [pollCounter, setPollCounter] = useState(0)
+  const { data } = useQuery({
+    queryKey: ['registration-status-polling'],
+    queryFn: async () => pollRegistrations(),
+    refetchInterval: REFETCH_INTERVAL,
+    enabled: processing,
+    onSuccess: () => {
+      setPollCounter(pollCounter + 1)
+    },
+  })
+  if (
+    data &&
+    (data.status.payment === 'initialized' ||
+      data.status.competing === 'incoming')
+  ) {
+    setProcessing(false)
+  }
+  return (
+    processing && (
+      <div>
+        <div>Your registration is processing...</div>
+        <div>{data && `Registration status: ${data.status.competing}`}</div>
+        <div>
+          {pollCounter > 3 && (
```

Comment:
Would it be more robust to have this in terms of time passed, rather than poll count, in case the frequency is changed in the future?

Reply:
good point and time passed still makes sense even if we switch to event based